test(3664): add unit tests for network permissions components

EtherWizard33 opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->


## **Description**
This PR adds comprehensive unit tests for components involved in managing network permissions.

### Key Changes
- NetworkSelectorList tests:
  - Snapshot tests for rendering verification.
  - Tests for network selection and right accessory rendering.
- NetworkConnectMultiSelector tests:
  - Simulate network permission update flows.
  - Validate handling of multi-select behavior and correct chain ID usage.
- ConnectionDetails tests:
  - Verify proper rendering and date formatting for network permission details.
- PermittedNetworksInfoSheet tests:
  - Ensure the info sheet renders the correct title, description, and action button.
- PermissionsManager tests:
  - Check that the component renders the empty state when no network permissions exist.


<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Contributes to:  https://github.com/MetaMask/MetaMask-planning/issues/3664

## **Manual testing steps**

1. run jest for each test file, npx jest <test-file-path>, or just yarn test:unit

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

EtherWizard33 committed:
test(PermissionsManager): add test for empty state rendering

- Added a test case to verify that the PermissionsManager component correctly renders
  the empty state when no permissions exist.
- Validates the presence of the session manager container and the appropriate empty state texts.

EtherWizard33 committed:
test(NetworkConnectMultiSelector): add comprehensive tests for network selection and permission updates

- Ensure proper rendering of the NetworkConnectMultiSelector component.
- Verify initialization with permitted chains from caveat data.
- Test user interactions for network selection and toggling the update button state.
- Confirm proper handling of network disconnects when no networks are selected.
- Validate error handling when retrieving permitted chains.

This commit enhances our test coverage and reinforces the component's robustness.

EtherWizard33 committed:
test(PermittedNetworksInfoSheet): add tests and test IDs constants

Add test coverage for PermittedNetworksInfoSheet component including:
- Create constants file for test IDs
- Add snapshot test
- Test title, description and button rendering
- Test container styles
- Test button interaction
- Add test IDs to component elements

EtherWizard33 committed:
test(ConnectionDetails): add unit tests for connection details modal

Tests cover rendering, date formatting, default states and user interactions

EtherWizard33 committed:
refactor(tests): improve TypeScript type definitions

- Add explicit types for mock functions and navigation props
- Remove unused imports across test files
- Improve mock component type definitions

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

EtherWizard33 committed:
fix(test): resolve Jest mock factory reference error

- Move forwardRef and useImperativeHandle inside mock factory
- Add proper TypeScript types for BottomSheet ref
- Follow Jest naming conventions for mock variables

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

EtherWizard33 committed:
test(NetworkSelectorList): add unit tests for network selector component

- Add test suite for NetworkSelectorList component
- Mock react-native Linking API
- Test rendering, network selection, and right accessory functionality
- Add snapshot test

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

EtherWizard33 committed:
test(account-connect-multiselector): add initial tests and update testID for account list

- Introduce a new test file for AccountConnectMultiSelector with snapshot tests,
  accounts list display, add account button press, and primary action button press.
- Update AccountConnectMultiSelector.tsx to add a testID prop to the AccountSelectorList,
  so it can be properly targeted by tests.
- Skip tests for select/unselect checkbox, disconnect action, and back button press for now.

EtherWizard33 committed:
style: lint

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

vinnyhoward reviewed (dismissed):
LGTM!

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

EtherWizard33 committed:
chore: remove sonar exclusions

tommasini commented on app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.test.tsx:
```diff
@@ -0,0 +1,238 @@
+    expect(mockOnUserAction).not.toHaveBeenCalled();
+  });
+
+  xit('handles back button press', () => {
```
```suggestion
  it('handles back button press', () => {
```

EtherWizard33 replied:
Thanks for noticing the xits, I had created these extra tests alongside the others, then excluded them temporarily, then forgot to circle back and remove them as originally planned. I did keep one of the extra 5 tests and remove the other 4, will commit it in the next commit.

tommasini commented on app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.test.tsx:
```diff
@@ -0,0 +1,238 @@
+    ).toBeDefined();
+  });
+
+  xit('handles select all checkbox correctly', () => {
```
```suggestion
  it('handles select all checkbox correctly', () => {
```

EtherWizard33 replied:
Thanks for noticing the xits, I had created these extra tests alongside the others, then excluded them temporarily, then forgot to circle back and remove them as originally planned. I did keep one of the extra 5 tests and remove the other 4, will commit it in the next commit.

tommasini commented on app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.test.tsx:
```diff
@@ -0,0 +1,238 @@
+    expect(mockOnSelectAddress).toHaveBeenCalledWith(['0x1234', '0x5678']);
+  });
+
+  xit('handles unselect all correctly', () => {
```
```suggestion
  it('handles unselect all correctly', () => {
```

tommasini commented on app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.test.tsx:
```diff
@@ -0,0 +1,238 @@
+    ).toBeDefined();
+  });
+
+  xit('shows disconnect button when no accounts are selected', () => {
```
```suggestion
  it('shows disconnect button when no accounts are selected', () => {
```

EtherWizard33 replied:
Thanks for noticing the xits, I had created these extra tests alongside the others, then excluded them temporarily, then forgot to circle back and remove them as originally planned. I did keep one of the extra 5 tests will commit it in the next commit.

tommasini commented on app/components/Views/AccountConnect/AccountConnectMultiSelector/AccountConnectMultiSelector.test.tsx:
```diff
@@ -0,0 +1,238 @@
+    expect(getByTestId(ConnectedAccountsSelectorsIDs.DISCONNECT)).toBeDefined();
+  });
+
+  xit('handles disconnect all action', () => {
```
```suggestion
  it('handles disconnect all action', () => {
```

EtherWizard33 replied:
Thanks for noticing the xits, I had created these extra tests alongside the others, then excluded them temporarily, then forgot to circle back and remove them as originally planned. I did keep one of the extra 5 tests will commit it in the next commit.

EtherWizard33 committed:
test: remove obsolete tests from AccountConnectMultiSelector

Remove skipped tests for the select/unselect all checkboxes, disconnect all action,
and back button press. Also, re-enable the disconnect button test when no accounts are selected.

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

EtherWizard33 committed:
test: update snapshots

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions

tommasini commented:
Lets merge main and run the e2e again please

EtherWizard33 committed:
Merge branch 'main' into test-3664-add-unit-test-coverage-to-multichain-permisisions
